### DISCUSSION TITLE
How to enable fzf history?

### DISCUSSION THREAD

**Discussion by Dronakurl** [LazyVim/LazyVim#4127] [Q&A]
I want to use fzf-lua and use the history of fzf.
Here is an explanation, but I cannot figure it out, how to set this up:
https://github.com/ibhagwan/fzf-lua/wiki#custom-history

I added the following to my config:

```lua
  {
    "ibhagwan/fzf-lua",
    opts = function(_, opts)
      return vim.tbl_deep_extend("force", opts, {
        keymap = {
          fzf = {
            ["ctrl-f"] = "next-history",
            ["ctrl-b"] = "prev-history",
          },
        },
   }
}
```

But ctrl-b did not do anything. Can anyone please help? I often grep stuff and want to go back to my previous search

**Comment by Dronakurl** [comment]
Edit: This seems to work. Apparently, the fzf keymap is masked by lua

```lua
  {
    "ibhagwan/fzf-lua",
    opts = function(_, opts)
      return vim.tbl_deep_extend("force", opts, {
        keymap = {
          fzf = {
            ["down"] = "next-history",
            ["up"] = "prev-history",
          },
        },
        files = {
          fzf_opts = {
            ["--history"] = vim.fn.stdpath("data") .. "/fzf-lua-files-history",
          },
          cwd_prompt = false,
        },
        grep = {
          fzf_opts = {
            ["--history"] = vim.fn.stdpath("data") .. "/fzf-lua-grep-history",
          },
        },
```

**Comment by murar8** [comment]
This also works for me:
```lua
return {
	"ibhagwan/fzf-lua",
	opts = {
		fzf_opts = {
			["--history"] = vim.fn.stdpath("data") .. "/fzf-history",
		},
	},
}
```

**Reply by FilipHarald** [reply]
To add to this, it also adds the default key mapping `ctrl-p` and `ctrl-n` for browsing history. (And they could be customized using `prev-history` and `next-history`, https://github.com/junegunn/fzf/blob/687074e77290f699199bce92c7541bbc2890a7cd/src/options.go#L1777)
